app/controllers/searches_controller.rb

```diff
@@ -1,2 +1,33 @@
 class SearchesController < ApplicationController
+  INSTAGRAM_SEARCH_USERS_URI = "https://api.instagram.com/v1/users/search?client_id=#{ENV["INSTAGRAM_CLIENT_ID"]}&"
+
+  def search
+    if params[:search_twitter]
+      if params[:search_twitter].empty?
+        redirect_to root_path
+      else
+        redirect_to search_results_path("twitter", params[:search_twitter])
+      end
+    end
+
+    # OPTIMIZE: refactor so there are less nested if's
```

Comment:
I was about to say the same thing. Could you rename the field from `search_twitter` and `search_instagram` to just `search` or similar? Then you could pass along the provider as a hidden field in the search form. Then you could do something like...

```ruby
def search
  redirect_to root_path if params[:search].empty? # empty checks both .nil? and .blank?
  redirect_to search_results_path(params[:provider], params[:search])
end
```